tests/sles4sap/sapconf.pm

```diff
@@ -104,14 +105,24 @@ sub verify_sapconf_service {
     die "Command 'systemctl status $svc' output is not recognized" unless ($output =~ m|$active|s or $output =~ m|$success|s);
 }
 
+sub test_profile {
+    my $profile = shift;
+    assert_script_run "tuned-adm profile_info $profile" if is_sle('15+');
+    assert_script_run "tuned-adm profile $profile";
+    sleep 4;    # add a small sleep to ensure that tuned-adm has correctly switched profile
+    check_profile($profile);
+}
+
 sub run {
     my ($self) = @_;
-    my @tuned_profiles = is_sle('>=15') ?
-      qw(balanced desktop latency-performance network-latency network-throughput
-      powersave sapconf saptune throughput-performance virtual-guest virtual-host)
-      : qw(balanced desktop latency-performance network-latency
-      network-throughput powersave sap-ase sap-bobj sap-hana sap-netweaver
-      throughput-performance virtual-guest virtual-host);
+    my @tuned_profiles = qw(balanced desktop latency-performance network-latency network-throughput
+      powersave throughput-performance virtual-guest);
+    # Testing 'virtual-host' on a VM isn't very useful and can lead to sporadic timeout issues
+    push @tuned_profiles, 'virtual-host' unless (check_var('BACKEND', 'qemu'));
+    # Add some profiles depending of the OS version
+    push @tuned_profiles, is_sle('15+') ? qw(sapconf saptune) : qw(sap-ase sap-bobj sap-hana sap-netweaver);
+    # Sort the array for the next regexp check
+    @tuned_profiles = sort @tuned_profiles;
```

Comment:
I think we can move this to L142, and that way we can use a single loop between lines L156 and L162. Sorry didn't notice it before merging.